### ISSUE TITLE
QBMEvent doesn't return correct date

### ISSUE THREAD

**Issue by neshyurik** [QuickBlox/quickblox-ios-sdk#494]
I'm using Quickblox to send remote notifications via APNs from iPhones using the Quickblox iOS SDK.
When I create event with type with fixed date it  responses the event but for some time in the future (Example: 2084-06-17 2:28:50 PM +0000)

```objectivec
  [QBRequest userWithLogin:self.currentClipon.receiverName successBlock:^(QBResponse *response, QBUUser *user) {
       
        NSTimeInterval oneMinute = 60.0;
        NSDate *dateToSend = [NSDate dateWithTimeIntervalSinceNow:oneMinute];
        
        QBMEvent *event = [QBMEvent event];
        event.notificationType = QBMNotificationTypePush;
        event.type = QBMEventTypeFixedDate;
        event.date = dateToSend;
        event.pushType = QBMPushTypeAPNS;
        
        event.usersIDs = [NSString stringWithFormat:@"%lu", (unsigned long)user.ID];
        event.isDevelopmentEnvironment = ![QBApplication sharedApplication].productionEnvironmentForPushesEnabled;

        // custom params
        NSMutableDictionary  *dictPush=[NSMutableDictionary  dictionaryWithObjectsAndKeys:@"Qwerty", @"message", nil];
        [dictPush setObject:@"beep-warmguitar.aif" forKey:@"ios_sound"];
        
        NSError *error = nil;
        NSData *sendData = [NSJSONSerialization dataWithJSONObject:dictPush options:NSJSONWritingPrettyPrinted error:&error];
        NSString *jsonString = [[NSString alloc] initWithData:sendData encoding:NSUTF8StringEncoding];
        
        event.message = jsonString;
        
        [QBRequest createEvent:event successBlock:^(QBResponse *response, NSArray *events) {
            
        }
                    errorBlock:^(QBResponse *response) {
                        
                    }];
        
    } errorBlock:^(QBResponse *response) {
        
    }];
```

**Environment details**
IOS 9.3.1 (Deployment target -8.0)
Xcode Version 7.3.1

**Did this work before?**
No

**Logs**
```
<__NSArrayI 0x150a5cce0>(

  [QBMEvent]:
  ID:7178125
  created at:2016-05-30 11:14:42 AM +0000
  updated at:2016-05-30 11:14:42 AM +0000
  active:YES
  notificationType:push
  pushType:apns
  usersIDs:13126192
  usersExternalIDs:(null)
  usersTagsAny:(null)
  usersTagsAll:(null)
  usersTagsExclude:(null)
  name:(null)
  isEnvironmentDevelopment:YES
  message:payload=eyJhcHMiOnsiYWxlcnQiOiJ7XG4gIFwibWVzc2FnZVwiIDogXCJRd2VydHlcIixcbiAgXCJpb3Nfc291bmRcIiA6IFwiYmVlcC13YXJtZ3VpdGFyLmFpZlwiXG59Iiwic291bmQiOiJkZWZhdWx0In19
  type:fixed_date
  date:2084-06-17 2:28:50 PM +0000
  endDate:(null)
  period:0
  occuredCount:0
  senderID:13125511
)
```

**Comment by soulfly** [comment]
```
Response JSON: {
    event =     {
        active = 1;
        "application_id" = 41443;
        "created_at" = "2016-06-03T13:21:42Z";
        date = 4743931296260895225;
        "end_date" = "<null>";
        "event_type" = "fixed_date";
        id = 7230684;
        kind = API;
        message = "payload=eyJhcHMiOnsiYWxlcnQiOiJ7XG4gIFwibWVzc2FnZVwiIDogXCJRd2VydHlcIixcbiAgXCJpb3Nfc291bmRcIiA6IFwiYmVlcC13YXJtZ3VpdGFyLmFpZlwiXG59Iiwic291bmQiOiJkZWZhdWx0In19";
        name = "<null>";
        "notification_channel" =         {
            name = apns;
        };
        "occured_count" = 0;
        period = "<null>";
        "subscribers_selector" =         {
            environment = development;
            "tags_query" = "<null>";
            "user_ids" =             (
                13261027,
                13261027
            );
        };
        "updated_at" = "2016-06-03T13:21:42Z";
        "user_id" = 13284432;
    };
}
4:25 Response headers: {
    "Access-Control-Allow-Origin" = "*";
    "Cache-Control" = "max-age=0, private, must-revalidate";
    Connection = "keep-alive";
    "Content-Length" = 592;
    "Content-Type" = "application/json; charset=utf-8";
    Date = "Fri, 03 Jun 2016 13:21:42 GMT";
    Etag = "\"78b9823697ea5a5ed46d0aff10e01689\"";
    "QB-Token-ExpirationDate" = "2016-06-03 15:21:05 UTC";
    "QuickBlox-REST-API-Version" = "0.1.1";
    Server = "nginx/1.8.1";
    Status = "201 Created";
    "Strict-Transport-Security" = "max-age=15768000;";
    "X-Rack-Cache" = "invalidate, pass";
    "X-Request-Id" = b6e9a13d391a4e81b3682c20f4718aac;
    "X-Runtime" = "0.237856";
    "X-UA-Compatible" = "IE=Edge,chrome=1";
}

```

**Comment by Raikerian** [comment]
Hi @neshyurik 

Can you try to set the date without seconds?

Use something like this:
```objectivec
NSTimeInterval oneMinute = 60.0;
NSDate *dateToSend = [NSDate dateWithTimeIntervalSinceNow:oneMinute];
NSTimeInterval timeSince1970 = [dateToSend timeIntervalSince1970];
timeSince1970 -= fmod(timeSince1970, 60); // subtract away any extra seconds
NSDate *nowMinus = [NSDate dateWithTimeIntervalSince1970:timeSince1970];
```

Right now I'm investigating the issue and will post update on whats wrong as soon as possible.

**Comment by Raikerian** [comment]
I was able to figure it out and fix. It will be available in the next release.

The problem was in wrong response serialization and wrong request serialization if date had milliseconds.
Thanks for reporting once again.

**Comment by neshyurik** [comment]
Thanks a lot! Waiting for new release. Good luck)

**Comment by Raikerian** [comment]
Fixed in 2.7.3.
